src/core/jstick.c

```diff
@@ -154,6 +154,19 @@ void js_guid_create(_js_device *jdev) {
 		(*(word + 5)) = jdev->usb.product_id - 300;
 		(*(word + 6)) = jdev->usb.version;
 		(*(word + 7)) = jdev->usb.version - 400;
+#if defined (_linux__)
+		if (ustrlen(jdev->uniq)) {
```

Comment:
Not necessary, because
`for (const char *s = jdev->uniq; *s; ++s) {`
already does that

Reply:
You are definitely right.

Reply:
[cd5df74](https://github.com/punesemu/puNES/pull/407/commits/cd5df745453bd8794b33a0d249577b56eedb1b91)